src/OrchardCore/OrchardCore.DisplayManagement/Zones/ZoneShapes.cs

```diff
@@ -13,43 +14,70 @@ namespace OrchardCore.DisplayManagement.Zones
     [Feature(Application.DefaultFeatureId)]
     public class ZoneShapes : IShapeAttributeProvider
     {
+        // By convention all placement delimiters default to the name 'Content' when not specified during placement.
+        private const string ContentKey = "Content";
+
         [Shape]
-        public async Task<IHtmlContent> Zone(dynamic DisplayAsync, dynamic Shape)
+        public async Task<IHtmlContent> Zone(IDisplayHelper DisplayAsync, IEnumerable<object> Shape)
         {
             var htmlContentBuilder = new HtmlContentBuilder();
             foreach (var item in Shape)
             {
-                htmlContentBuilder.AppendHtml(await DisplayAsync(item));
+                htmlContentBuilder.AppendHtml(await DisplayAsync.ShapeExecuteAsync(item));
             }
 
             return htmlContentBuilder;
         }
 
         [Shape]
-        public async Task<IHtmlContent> ContentZone(dynamic DisplayAsync, dynamic Shape, IShapeFactory ShapeFactory)
+        public async Task<IHtmlContent> ContentZone(IDisplayHelper DisplayAsync, dynamic Shape, IShapeFactory ShapeFactory)
         {
             var htmlContentBuilder = new HtmlContentBuilder();
 
-            var shapes = ((IEnumerable<dynamic>)Shape);
+            // This maybe a collection of IShape, IHtmlContent, or plain object.
+            var shapes = ((IEnumerable<object>)Shape);
+
+            // Evaluate shapes for grouping metadata, when it is not an IShape it cannot be grouped.
+            var isGrouped = shapes.Any(x => x is IShape s &&
+                (!String.IsNullOrEmpty(s.Metadata.Tab) ||
+                !String.IsNullOrEmpty(s.Metadata.Card) ||
+                !String.IsNullOrEmpty(s.Metadata.Column)));
 
-            var groupings = shapes.GroupBy(x =>
+            // When there is no grouping metadata on any shapes just render the Zone.
+            if (!isGrouped)
             {
-                // By convention all placement delimiters default to the name 'Content' when not specified during placement.
-                var key = (string)x.Metadata.Tab;
-                if (String.IsNullOrEmpty(key))
+                foreach (var item in shapes)
                 {
-                    key = "Content";
+                    htmlContentBuilder.AppendHtml(await DisplayAsync.ShapeExecuteAsync(item));
                 }
 
-                // Remove any positioning modifier.
-                var modifierIndex = key.IndexOf(';');
-                if (modifierIndex != -1)
+                return htmlContentBuilder;
+            }
```

Comment:
Noop grouping here and just render the shapes instead of creating all the group shapes.